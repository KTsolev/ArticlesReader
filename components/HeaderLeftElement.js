import React from "react";
import { View, TouchableOpacity } from "react-native";
import { Icon } from 'react-native-elements';
import { showHideFiltersMenu } from '../actions/actions';
import { connect } from 'react-redux';
import FiltersMenu from './FiltersMenu';

class HeaderLeftElement extends React.Component {
   render(){ 
       return <View style={{ 
           flexDirection: 'row',
           justifyContent: 'flex-start',
           padding: 5, 
           maxWidth: 50}}>
       <TouchableOpacity onPress={() => this.props.toggleFilters(!this.props.showFilters)}>
        <Icon
            name='bars' 
            size={20} 
            type='font-awesome'
            color='#fff' />
       </TouchableOpacity>
       <FiltersMenu />
     </View>
    }
}

const mapStateToProps = (state)=> {
    return {
        showFilters: state.uiElementsState.showFilters
    }
}
    
const mapDispatchToProps = dispatch => {
    return {
        toggleFilters: (toggle) => {
            dispatch(showHideFiltersMenu(toggle));
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(HeaderLeftElement);
  
 