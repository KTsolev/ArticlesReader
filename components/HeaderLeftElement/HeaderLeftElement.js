import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Icon } from 'react-native-elements';
import { showHideFiltersMenu } from '../../actions/actions';
import { connect } from 'react-redux';
import { styles } from './HeaderLeftElementStyles';

class HeaderLeftElement extends React.Component {
    render(){
       const { container } = styles; 
       return <View style={container}>
       <TouchableOpacity onPress={() => this.props.toggleFilters(!this.props.showFilters)}>
        <Icon
            name='bars' 
            size={20} 
            type='font-awesome'
            color='#fff' />
       </TouchableOpacity>
     </View>
    }
}

const mapStateToProps = (state)=> {
    return {
        showFilters: state.uiElementsState.showFilters
    }
}
    
const mapDispatchToProps = dispatch => {
    return {
        toggleFilters: (toggle) => {
            dispatch(showHideFiltersMenu(toggle));
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(HeaderLeftElement);
  
 