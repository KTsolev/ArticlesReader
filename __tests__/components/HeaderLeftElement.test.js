import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import { TouchableOpacity } from 'react-native'
import { render, fireEvent } from 'react-native-testing-library';
import configureMockStore from 'redux-mock-store';
import HeaderLeftElement from '../../components/HeaderLeftElement/HeaderLeftElement';

const createTestProps = (props) => ({
    navigation: {
      navigate: jest.fn(),
      getParam: (name) => {
          if (name==='url') {
            return 'https://jestjs.io/';
          } 
          return null;
      }
    },
    ...props
  });

  describe('<HeaderLeftElement />', () => {
    let props;
    let middlewares;
    let mockStore;
    
    beforeEach(() => {
      props = createTestProps({});
      middlewares = [thunk]
      mockStore = configureMockStore(middlewares)
      
      props.store = mockStore({
          uiElementsState: {
              showFilters: false,
              showPopUpMenu: false
          },
          articles: {
              filters: {}
          }
      });
    });
    
    test('renders the component', () => {
        const wrapper = shallow(<HeaderLeftElement {...props} />);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });

    test('Dispatches the correct action and payload, when filters menu is opened', () => {
      const { queryAllByType } = render(<HeaderLeftElement {...props} />);
      fireEvent.press(queryAllByType(TouchableOpacity)[0]);
      const expectedActions = [
          {
            payload: {
              toggle: true,
            },
            type: 'TOGGLE_FILTERS_OPTIONS',
          },
        ];
    
        expect(props.store.getActions()).toEqual(expectedActions);
    });
});
