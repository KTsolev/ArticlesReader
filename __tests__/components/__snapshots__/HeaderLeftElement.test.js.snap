// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLeftElement /> renders the component 1`] = `
<HeaderLeftElement
  navigation={
    Object {
      "getParam": [Function],
      "navigate": [MockFunction],
    }
  }
  showFilters={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  toggleFilters={[Function]}
/>
`;
